src/components/molecules/BranchSelect/BranchCell.tsx

```diff
@@ -20,22 +23,25 @@ const BranchCell: React.FC<IProps> = props => {
 
   const dispatch = useAppDispatch();
   const currentBranch = useAppSelector(state => state.git.repo?.currentBranch);
-  const selectedProjectRootFolder = useAppSelector(state => state.config.selectedProjectRootFolder);
+  const selectedProjectRootFolder = useAppSelector(state => state.config.selectedProjectRootFolder) || '';
 
   const deleteBranch = (e: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
     e.stopPropagation();
 
     Modal.confirm({
       title: `Are you sure you want to delete ${branch.name}?`,
       onOk() {
-        promiseFromIpcRenderer('git.deleteLocalBranch', 'git.deleteLocalBranch.result', {
-          localPath: selectedProjectRootFolder,
-          branchName: branch.name,
-        }).then(result => {
-          if (result.error) {
-            showGitErrorModal(`Deleting ${branch.name} failed`, `git branch -d ${branch.name}`, dispatch);
-          }
-        });
+        dispatch(setGitLoading(true));
+
+        try {
+          deleteLocalBranch({localPath: selectedProjectRootFolder, branchName: branch.name});
+          dispatch(
+            setAlert({type: AlertEnum.Success, title: `Branch ${branch.name} deleted successfully`, message: ''})
+          );
```

Comment:
maybe here it makes sense to do something like:
```suggestion
          deleteLocalBranch({localPath: selectedProjectRootFolder, branchName: branch.name}).unwrap().then(() => {
          dispatch(
            setAlert({type: AlertEnum.Success, title: `Branch ${branch.name} deleted successfully`, message: ''})
          );
          });
```